cocotb/log.py

```diff
@@ -65,89 +70,43 @@ def __init__(self, name):
         else:
             hdlr.setFormatter(SimLogFormatter())
             self.colour = False
-        self._cache = {}
-        self.name = name
-        self.handlers = []
-        self.disabled = False
-        self.filters = []
+
         self.propagate = False
-        logging.__init__(name)
         self.addHandler(hdlr)
-        self.setLevel(logging.NOTSET)
-
-""" Need to play with this to get the path of the called back,
-    construct our own makeRecord for this """
 
-
-class SimLog(object):
-    def __init__(self, name, ident=None):
-        self._ident = ident
-        self._name = name
-        self.logger = logging.getLogger(name)
-        if self._ident is not None:
-            self._log_name = "%s.0x%x" % (self._name, self._ident)
-        else:
-            self._log_name = name
-
-    def _makeRecord(self, level, msg, args, extra=None):
-        if self.logger.isEnabledFor(level):
-            frame = inspect.stack()[2]
-            info = inspect.getframeinfo(frame[0])
-            record = self.logger.makeRecord(self._log_name,
-                                            level,
-                                            info.filename,
-                                            info.lineno,
-                                            msg,
-                                            args,
-                                            None,
-                                            info.function,
-                                            extra)
-            self.logger.handle(record)
-
-    def _willLog(self, level):
-        """ This is for user from the C world
-            it allows a check on if the message will
-            be printed. Saves doing lots of work
-            for no reason.
+    def _log_from_c(self, level, filename, lineno, msg, function):
```

Comment:
Looks like logging uses lowerCamelCase for function names. Should we do the same here for consistency?

Reply:
Since I need to edit the message below anyway, why not